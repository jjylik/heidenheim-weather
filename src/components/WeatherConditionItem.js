import React from "react";
import { useSpring, animated } from "react-spring";
import WeatherIcon from './WeatherIcon'


const WeatherConditionItem = ({
  delay = 0,
  label,
  value,
  icon,
  children,
  round = val => val
}) => {
  const numberAnimation = useSpring({
    delay: delay + 500,
    number: value,
    from: { number: 0 }
  });
  const opacityAnimation = useSpring({
    opacity: 1,
    delay: delay,
    from: { opacity: 0 }
  });
  return (
    <animated.div style={opacityAnimation} className="weather-row">
      <WeatherIcon iconName={icon} />
      <span>{label}</span>
      <div>
        <animated.span>
          {numberAnimation.number.interpolate(round)}
        </animated.span>
        {children}
      </div>
    </animated.div>
  );
};

export default WeatherConditionItem;