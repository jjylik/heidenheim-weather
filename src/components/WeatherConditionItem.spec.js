import React from 'react';
import WeatherConditionItem from './WeatherConditionItem';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(<WeatherConditionItem delay={200} label="Rain chance" value={200} ><span>Hello</span></WeatherConditionItem>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});