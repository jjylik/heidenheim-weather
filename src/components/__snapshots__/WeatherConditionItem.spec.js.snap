// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="weather-row"
  style={
    Object {
      "opacity": 0,
    }
  }
>
  <i
    className="wi undefined"
  />
  <span>
    Rain chance
  </span>
  <div>
    <span>
      0
    </span>
    <span>
      Hello
    </span>
  </div>
</div>
`;
